# Java 中的异常处理

> 原文:[https://www . software testing material . com/exception-handling-in-Java/](https://www.softwaretestingmaterial.com/exception-handling-in-java/)

Java 编程语言使用异常来处理错误和其他异常事件。

**什么是例外:**

异常是中断程序指令正常流程的事件。异常发生在程序执行期间，并终止程序。如前所述，Java 语言使用异常来处理错误和其他异常事件。

当方法中出现错误时，该方法创建一个对象(即异常对象)并将其交给运行时系统。这个异常对象包含关于错误的信息。创建异常对象并将其处理给运行时系统的过程称为抛出异常。每当一个方法抛出一个异常，运行时系统就试图找到一些异常处理程序来处理这个异常。在处理异常的过程中，我们必须提供一个用户友好的消息，让用户容易理解，而不是简单地抛出非技术人员难以理解的系统生成消息。

```java
Exception in thread "main" java.lang.ArithmeticException: / by zero
	at classExceptionHandling.UncheckedExceptionClass.main(UncheckedExceptionClass.java:15)
```

异常处理的主要目的是继续程序的流程。

**异常类型:**

有两种类型的例外:

1.已检查的异常
2。未检查的异常

**1。已检查异常:**

检查异常发生在编译时。这些也称为编译时异常。作为程序员，我们必须在编译时处理这些异常。

让我们看一个例子:

```java
package classExceptionHandling;

import java.io.FileInputStream;

public class CheckedExceptionClass {

	public static void main(String [] args){

		FileInputStream fis = new FileInputStream("D:/TestFile.txt");			

	}

}
```

输出:

```java
Exception in thread "main" java.lang.Error: Unresolved compilation problem: 
	Unhandled exception type FileNotFoundException

	at classExceptionHandling.CheckedExceptionClass.main(CheckedExceptionClass.java:10)
```

让我们看看如何处理这个问题。

使用 throws 关键字:

```java
package classExceptionHandling;

import java.io.FileInputStream;
import java.io.FileNotFoundException;

public class CheckedExceptionClass {

	public static void main(String [] args) throws FileNotFoundException{

		FileInputStream fis = new FileInputStream("D:/TestFile.txt");			

	}

}
```

使用 try catch 块:

```java
package classExceptionHandling;

import java.io.FileInputStream;

public class CheckedExceptionClass {

	public static void main(String [] args){

		try{
			FileInputStream fis = new FileInputStream("D:/TestFile.txt");			
		}catch (Exception e){
			System.out.println("File Not Found");
		}

	}

}
```

一些已检查的异常如下:

*   异常
*   ClassNotFoundException
*   处理

**2。未检查的异常:**

未检查的异常发生在执行时。这些也称为运行时异常。作为一名程序员，我们必须提前判断并妥善处理它们。

让我们看一个例子:

```java
package classExceptionHandling;

public class UncheckedExceptionClass {

	public static void main(String [] args){
		int a=100;
		int b=0;
		System.out.println(a/b);
	}

}
```

输出:

```java
Exception in thread "main" java.lang.ArithmeticException: / by zero
	at classExceptionHandling.UncheckedExceptionClass.main(UncheckedExceptionClass.java:8)
```

处理异常:

```java
package classExceptionHandling;

public class UncheckedExceptionClass {

	public static void main(String [] args){
		int a=100;
		int b=0;
		try {
			System.out.println(a/b);
		}catch (Exception e){
			System.out.println(e.getMessage());
		}
	}

}
```

输出:

```java
/ by zero
```

一些未检查的异常如下:

*   算术异常
*   空指针异常
*   ArrayIndexOutOfBoundsException
*   非法数据异常

> 必读: [Java 教程](https://www.softwaretestingmaterial.com/)